docs/cms.rst

```diff
@@ -0,0 +1,122 @@
+.. This Source Code Form is subject to the terms of the Mozilla Public
+.. License, v. 2.0. If a copy of the MPL was not distributed with this
+.. file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+.. _cms:
+
+====================
+CMS capability (WIP)
+====================
+
+From 2024, Bedrock's CMS will be powered by `Wagtail CMS`_.
+
+This page is currently a skeleton for documentation to be added as the work evolves.
+
+High-level summary
+------------------
+
+Wagtail CMS will be used to make either entire pages or portions of pages (let's
+call them 'surfaces') content-editable on www.mozilla.org. It is not a free-for-all
+add-any-page-you-like approach, but rather a careful rollout of surfaces with
+appropriate guard rails that helps ensure the integrity, quality and security of
+www.mozilla.org.
+
+Surfaces will be authored via a closed 'Editing' deployment and when those changes
+are published they will become visible on the main www.mozilla.org 'Web' deployment.
+
+Accessing the CMS
+-----------------
+
+Details to come
+
+
+Editing current content surfaces
+--------------------------------
+
+Official `Editor Guide`_.
+
+Bedrock-specific details to come.
+
+
+
+Adding new content surfaces
+---------------------------
+
+Official `Editor Guide`_.
+
+Bedrock-specific details to come.
+
+
+Images
+------
+
+Using editor-uploaded images in templates
+=========================================
+
+Images may be uploaded into Wagtail's Image library and then included in
+content-managed surfaces that have fields/spaces for images.
+
+Images are stored in the same media bucket that fixed/hard-coded Bedrock
+images get put in, and coexist alongside them, being namespaced into a
+directory called ``custom-media/``.
+
+If a surface uses an image, image use must be made explicit via template markup
+— we need to state both *where* and *how* an image will be used in the template,
+including specifying the size the image will be. This is because — by design
+and by default — Wagtail can generate any size version that the template
+mentions by providing a "filter spec" e.g.
+
+.. code-block:: jinja
+
+    {% set background=image(page.product_image, "max-1024x1024") %}
+    <img class="some-class" src="{{ product_image.url }})"/>
```

Comment:
**Question:** this is probably a wagtail thing, but why do we need to set a variable called `background` if it's not used directly? (I also wonder if background is maybe a bit of a confusing example, since background images are usually applied as CSS background images.

Reply:
Ugh that was me repurposing an example and not cleaning it properly. Will fix now